client/src/app/pages/issues/issue-detail-drawer/file-incidents-detail-modal/file-incidents-detail-modal.tsx

```diff
@@ -121,30 +123,26 @@ export const FileIncidentsDetailModal: React.FC<
                   </Grid>
                 ) : null}
               </Tab>
-            ))}
-          </Tabs>
-          {totalNumIncidents > 5 ? (
-            <Tabs
-              activeKey="all"
-              onSelect={(_event, tabKey) =>
-                setActiveTabIncidentId(tabKey as IncidentIdOrAll)
-              }
-            >
-              <Tab
-                eventKey="all"
-                title={`All incidents (${totalNumIncidents})`} // TODO i18n
-              >
-                <Alert
-                  isInline
-                  variant="info"
-                  className={spacing.mtMd}
-                  title="TODO"
-                />
-                <FileAllIncidentsTable fileReport={fileReport} />
-              </Tab>
-            </Tabs>
-          ) : null}
-        </>
+            )),
+            ...(totalNumIncidents > 5
+              ? [
+                  <Tab
+                    key="all"
+                    eventKey="all"
+                    title={`All incidents (${totalNumIncidents})`} // TODO i18n
+                  >
+                    <Alert
+                      isInline
+                      variant="info"
+                      className={spacing.mtMd}
+                      title="TODO"
+                    />
+                    <FileAllIncidentsTable fileReport={fileReport} />
+                  </Tab>,
+                ]
+              : []),
+          ]}
```

Comment:
As an alternative to the array spread (which does look ok here), you could throw in a `.flat().filter(Boolean)` and not worry about it:

```tsx
  {[
    firstFiveIncidents.map((incident, index) => ( /* snip */ )),
    totalNumIncidents > 5 && [ /* snip */ ],
  ].flat(1).filter(Boolean)}
```

Reply:
Ah, that's a good idea! Nice and clean.